lib/contracts/notes/game/genkey.masm

```diff
@@ -0,0 +1,48 @@
+use.std::math::ecgfp5::group
+use.miden::account
+use.miden::note
+use.miden::contracts::wallets::basic->wallet
+
+const.G=7
+const.MASKING_FACTOR_SLOT=1
+const.PUBLIC_KEY_SLOT=2
+
+proc.gen_privatekey
+    # => [r, ...]
+    dup push.2 gt assert
+    dup mem_store.0
+    push.G swap exp
+    # => [G^r]
+
+    padw drop push.PUBLIC_KEY_SLOT
+    # => [PUBLIC_KEY_SLOT, 0, 0, 0, G^r]
+    exec.account::set_item
+    dropw dropw
+
+    mem_load.0 
+    # => [r]
+    padw drop push.MASKING_FACTOR_SLOT
+    # => [MASKING_FACTOR_SLOT, 0, 0, 0, r]
```

Comment:
Just to be semantically correct if we are using `r` to calculate the public key then it's not a masking factor. it's secret key (maybe you should change the naming) and slot for masking factor should be different.

Reply:
Maybe at the point of generation itself we can keep set the masking factor as well for now for testing purpose secret key for all the players would be same just to ignore pub key aggregation for now.

Reply:
My bad, it was written r everywhere so got confused.